dvc/commands/studio.py

```diff
@@ -0,0 +1,171 @@
+import argparse
+import os
+
+from funcy import get_in
+
+from dvc.cli.utils import append_doc_link, fix_subparsers
+from dvc.commands.config import CmdConfig
+from dvc.log import logger
+
+logger = logger.getChild(__name__)
+
+
+class CmdStudioLogin(CmdConfig):
+    def run(self):
+        from dvc_studio_client.auth import StudioAuthError, get_access_token
+
+        from dvc.env import DVC_STUDIO_URL
+        from dvc.ui import ui
+        from dvc.utils.studio import STUDIO_URL
+
+        name = self.args.name
+        hostname = self.args.hostname or os.environ.get(DVC_STUDIO_URL) or STUDIO_URL
+        scopes = self.args.scopes
+
+        try:
+            token_name, access_token = get_access_token(
+                token_name=name,
+                hostname=hostname,
+                scopes=scopes,
+                use_device_code=self.args.use_device_code,
+                client_name="dvc",
+            )
+        except StudioAuthError as e:
+            ui.error_write(str(e))
+            return 1
```

Comment:
Note to self: I need to look into `dvc-studio-client` if we can improve upon the API, especially around the side effects of opening a browser and printing things.

Reply:
Also I am having a second thought if webbrowser.open() belongs in dvc instead. :(